ctl/src/command.rs

```diff
@@ -410,14 +407,14 @@ pub fn upgrade_worker(mut channel: Channel<CommandRequest,CommandResponse>, time
   });
 
   if timeout > 0 && recv.recv_timeout(Duration::from_millis(timeout)).is_err() {
-    eprintln!("Command timeout. The proxy didn't send answer");
-    exit(1);
+    bail!("Command timeout. The proxy didn't send answer");
   }
 
-  timeout_thread.join().expect("upgrade_worker: Timeout thread should correctly terminate")
+  Ok(timeout_thread.join().expect("upgrade_worker: Timeout thread should correctly terminate"))
```

Comment:
Still an expect here, is it intended?

Reply:
Can't use `expect` or `?` witch `std::thread::JoinHandle::join()`, unfortunately

Reply:
Maybe you can match it instead?

Reply:
alas, `join()` yields an error type that doesn't implement the traits needed for anyhow to trickle up the error :/

Reply:
You could use a if like:

```rust
if let Err(err) = timeout_thread.join() {
  return Err(anyhow::Error::from(format!("upgrade worker: thread timeout exceeded on join, {}", err));
}

Ok(())

Reply:
Got away with 

```rust
  timeout_thread.join().map_err(|error| {
    anyhow::Error::msg(
      format!("upgrade worker: thread timeout exceeded on join, {:?}", error)
    )
  })
```